drivers/staging/vc04_services/bcm2835-audio/bcm2835.c

```diff
@@ -312,6 +313,45 @@ static int snd_add_child_devices(struct device *device, u32 numchans)
 	return 0;
 }
 
+static u8 get_hdmi_connections(struct device *dev)
+{
+	struct device_node *firmware_node;
+	struct rpi_firmware *firmware;
+	u32 num_displays, i, display_id;
+	u8 bits = 0;
+	int ret;
+
+	firmware_node = of_parse_phandle(dev->of_node, "brcm,firmware", 0);
+	firmware = rpi_firmware_get(firmware_node);
+
+	if (!firmware)
+		return 0;
+
+	of_node_put(firmware_node);
+
+	ret = rpi_firmware_property(firmware,
+				    RPI_FIRMWARE_FRAMEBUFFER_GET_NUM_DISPLAYS,
+				    &num_displays, sizeof(u32));
+
+	if (ret)
+		return 0;
+
+	for (i = 0; i < num_displays; i++) {
+		display_id = i;
+		ret = rpi_firmware_property(firmware,
+				RPI_FIRMWARE_FRAMEBUFFER_GET_DISPLAY_ID,
+				&display_id, sizeof(display_id));
+		if (!ret) {
+			if (display_id == 2)
+				enable_hdmi0 = true;
+			if (display_id == 7)
+				enable_hdmi1 = true;
+		}
+	}
+
+	return bits;
```

Comment:
Is bits meant to be something other than 0?

Reply:
Good spot, left something in from a previous iteration of the code. Will sort when I get in to the office.

Reply:
Fixed up and renamed the function to be more accurate.